src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -403,8 +458,17 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
     disk = Disk(location=location, creation_data=creation_data, tags=(tags or {}),
                 sku=_get_sku_object(cmd, sku), disk_size_gb=size_gb, os_type=os_type, encryption=encryption)
 
+    if option == DiskCreateOption.empty and os_type:
+        # will set default value of hyper_v_generation and security_type
+        logger.warning('Starting Build 2023 event, az disk create command '
+                       'will deploy Trusted Launch VM by default. To know more about Trusted Launch, '
+                       'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
     if hyper_v_generation:
         disk.hyper_v_generation = hyper_v_generation
+        from azure.cli.command_modules.vm._vm_utils import is_os_disk
+        if hyper_v_generation == 'V1' and is_os_disk(DiskCreateOption, option, os_type):
+            # creating OS disk with Generation 1 will be recommended to upgrade security for user workloads
```

Comment:
Hello @AjKundnani Please help to take a look if the below notification message for **creating OS disk with '--hyper-v-generation V1'** meets your expectations.

**Please consider upgrading security for your VM resources by setting the hypervisor generation to V2. Please use "--hyper-v-generation V2" to set the hypervisor generation**

Reply:
> Hello @AjKundnani Please help to take a look if the below notification message for **creating OS disk with '--hyper-v-generation V1'** meets your expectations.
> 
> **Please consider upgrading security for your VM resources by setting the hypervisor generation to V2. Please use "--hyper-v-generation V2" to set the hypervisor generation**

@yanzhudd:
`Please consider upgrading security for your VM resources by using Gen 2 OS Disk and Trusted Launch security type. Please use "--hyper-v-generation V2" to set the hypervisor generation of disk. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch`